scripts/components/nav.component.js

```diff
@@ -20,12 +22,35 @@ export default class {
     this.qualDropdown = new Dropdown('qual', this.callbacks.onQualSelected);
     this.viewDropdown = new Dropdown('view', this.callbacks.onViewSelected);
 
+
     new BookmarkMenu();
     new Dropdown('bookmark', this.callbacks.onQuantSelected);
     new ShareMenu();
     new Dropdown('share');
   }
 
+  _setVars() {
+    this.el = document.querySelector('.c-nav');
+
+    this.AppNav = this.el.querySelector('.app-nav');
+    this.FlowsNav = this.el.querySelector('.flows-nav');
+    this.toggleBtn = this.el.querySelector('.js-toggle-menu');
+  }
+
+  _setEventListeners() {
+    this.toggleBtn.addEventListener('click', () => this.onToggleMenu());
+  }
+
+  onToggleMenu() {
+    this.toggleBtn.classList.toggle('open');
+    this.callbacks.onToggleAppMenu();
```

Comment:
Yeah, I agree with you. I did it because I didn't want to change the way the app communicates unless it's necessary. What's the requirements to do it this way? Communication between components and the possibility of need for sharing or similar?